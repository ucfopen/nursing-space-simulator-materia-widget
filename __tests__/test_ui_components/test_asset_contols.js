import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import AssetControls from "../../src/js/components/ui/asset_controls";

// Variables consistent against all tests
const selectedAsset = {
	title: "Bed-1"
};

const deselectMock = jest.fn();
const rotateMock = jest.fn();
const removeMock = jest.fn();

describe("Asset Controls Component", () => {
	it("renders asset controls", () => {
		const manipulationModeControlTree = renderer
			.create(
				<AssetControls
					manipulationMode={true}
					deselectAsset={deselectMock}
					rotateAsset={rotateMock}
					removeAsset={removeMock}
					selectedAsset={selectedAsset}
					currentX={1}
					currentZ={1}
				/>
			)
			.toJSON();

		expect(manipulationModeControlTree).toMatchSnapshot();

		const nonManipulationModeControlTree = renderer
			.create(
				<AssetControls
					manipulationMode={false}
					deselectAsset={deselectMock}
					rotateAsset={rotateMock}
					removeAsset={removeMock}
					selectedAsset={selectedAsset}
					currentX={1}
					currentZ={1}
				/>
			)
			.toJSON();

		expect(nonManipulationModeControlTree).toMatchSnapshot();

		expect(manipulationModeControlTree).not.toEqual(
			nonManipulationModeControlTree
		);
	});

	it("calls correct methods when clicking manupulation mode buttons", () => {
		let manipulationModeAssetControls = shallow(
			<AssetControls
				manipulationMode={"manipulation"}
				deselectAsset={deselectMock}
				rotateAsset={rotateMock}
				removeAsset={removeMock}
				selectedAsset={selectedAsset}
				currentX={1}
				currentZ={1}
			/>
		);
		manipulationModeAssetControls.find("#confirm").simulate("click");
		manipulationModeAssetControls.find("#rotate").simulate("click");
		manipulationModeAssetControls.find("#remove").simulate("click");

		expect(deselectMock).toBeCalled();
		expect(rotateMock).toBeCalledWith(1, 1);
		expect(removeMock).toBeCalledWith(1, 1);
	});

	it("calls correct methods while click while not in manipulation mode", () => {
		let nonManipulationModeAssetControls = shallow(
			<AssetControls
				manipulationMode={false}
				deselectAsset={deselectMock}
				rotateAsset={rotateMock}
				removeAsset={removeMock}
				selectedAsset={selectedAsset}
				currentX={1}
				currentZ={1}
			/>
		);

		nonManipulationModeAssetControls.find("#deselect").simulate("click");

		// Because deselect mock is global, it will be kept track of across all tests. Should be called twice by this point
		expect(deselectMock.mock.calls.length).toEqual(2);
	});
});
