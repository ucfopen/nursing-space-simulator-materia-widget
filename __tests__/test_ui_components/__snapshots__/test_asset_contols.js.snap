// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Controls Component renders asset controls 1`] = `
<div
  id="UI-selected-asset-options"
  onMouseDown={[Function]}
>
  <span
    id="selected-asset-label"
  >
    <span
      className="selected-asset-label-title"
    >
      Currently selected: 
      Bed-1
    </span>
  </span>
  <button
    id="deselect"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      enableBackground="new 0 0 125 125"
      height="55px"
      version="1.1"
      viewBox="0 0 125 125"
      width="55px"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g
        id="Layer_8"
      >
        <path
          d="M62.5,7.2C31.9,7.2,7.2,31.9,7.2,62.5s24.8,55.3,55.3,55.3s55.3-24.8,55.3-55.3S93.1,7.2,62.5,7.2z M62.5,26.7c5.2,0,10.1,1.1,14.6,3.1L29.8,77.1c-2-4.5-3.1-9.4-3.1-14.6C26.7,42.7,42.7,26.7,62.5,26.7z M62.5,98.3c-5.2,0-10.1-1.1-14.6-3.1l47.3-47.3c2,4.5,3.1,9.4,3.1,14.6C98.3,82.3,82.3,98.3,62.5,98.3z"
          fill="#DC7627"
          stroke="#000000"
          strokeMiterlimit="10"
          strokeWidth="3"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Asset Controls Component renders asset controls 2`] = `
<div
  id="UI-selected-asset-options"
  onMouseDown={[Function]}
>
  <span
    id="selected-asset-label"
  >
    <span
      className="selected-asset-label-title"
    >
      Currently selected: 
      Bed-1
    </span>
  </span>
  <button
    id="deselect"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      enableBackground="new 0 0 125 125"
      height="55px"
      version="1.1"
      viewBox="0 0 125 125"
      width="55px"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g
        id="Layer_8"
      >
        <path
          d="M62.5,7.2C31.9,7.2,7.2,31.9,7.2,62.5s24.8,55.3,55.3,55.3s55.3-24.8,55.3-55.3S93.1,7.2,62.5,7.2z M62.5,26.7c5.2,0,10.1,1.1,14.6,3.1L29.8,77.1c-2-4.5-3.1-9.4-3.1-14.6C26.7,42.7,42.7,26.7,62.5,26.7z M62.5,98.3c-5.2,0-10.1-1.1-14.6-3.1l47.3-47.3c2,4.5,3.1,9.4,3.1,14.6C98.3,82.3,82.3,98.3,62.5,98.3z"
          fill="#DC7627"
          stroke="#000000"
          strokeMiterlimit="10"
          strokeWidth="3"
        />
      </g>
    </svg>
  </button>
</div>
`;
